### PR TITLE
feat: Add Text Generation with Itex LSTM sample

### PR DESCRIPTION
**Pull request by nedo99**
# Adding a New Sample(s)

## Description

The 'Leveraging Intel Extension for TensorFlow with LSTM for Text Generation' sample demonstrates how to train your model with LSTM (Long short-term memory) faster by using Intel Extension for TensorFlow's LSTM training layer on Intel platform.

## Checklist
Administrative
- [x] Review sample design with the appropriate [Domain Expert](https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts): <insert Name Here>
- [ ] If you have any new dependencies/binaries, inform the oneAPI Code Samples Project Manager

Code Development
- [x] Implement coding guidelines and ensure code quality. [see wiki for details](https://github.com/oneapi-src/oneAPI-samples/wiki/General-Code-Guidelines)
- [x] Adhere to readme template 
- [ ] Enforce format via clang-format config file
- [x] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [x] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [x] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com

Security and Legal
- [ ] OSPDT Approval (see Project Manager for assistance)
- [ ] Compile using the following compiler flags and fix any warnings, the falgs are: "/Wall -Wformat-security -Werror=format-security"
- [ ] Bandit Scans (Python only)
- [ ] Virus scan

Review
- [ ] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [x] Review readme with Tom Lenth(@tomlenth) and/or Project Manager
- [ ] Tested using Dev Cloud when applicable

### CODE REVIEW

File: AI-and-Analytics/Features-and-Functionality/IntelTensorFlow_TextGeneration_with_LSTM/TextGenerationModelTraining.ipynb [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+    "\n",
+    "length = train_data_width + 1\n",
+    "lines = []\n",
+    "\n",
+    "for i in range(length, len(tokens)): \n",
+    "    seq = tokens[i - length:i]\n",
+    "    line = ' '.join(seq)\n",
+    "    lines.append(line)\n",
+    "\n",
+    "# Tokenization\n",
```
Comment (louie-tsai) [inline AI-and-Analytics/Features-and-Functionality/IntelTensorFlow_TextGeneration_with_LSTM/TextGenerationModelTraining.ipynb:305]: Could we put tokenization codes into a function, and just call a function here instead? those codes are little distracted for itex and keras comparison

Comment (nedo99) [reply AI-and-Analytics/Features-and-Functionality/IntelTensorFlow_TextGeneration_with_LSTM/TextGenerationModelTraining.ipynb:294]: Implemented with latest commit.

File: AI-and-Analytics/Features-and-Functionality/IntelTensorFlow_TextGeneration_with_LSTM/README.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+#### Running on the Command Line (Optional)
+
+1. Change to the sample directory.
+2. Run the script.
+   ```
+   python TextGenerationModelTraining.py
+   ```
+
+### Run the Sample on Intel® DevCloud (Optional)
```
[the remark this answers is not in the document]
Comment (nedo99) [reply AI-and-Analytics/Features-and-Functionality/IntelTensorFlow_TextGeneration_with_LSTM/README.md]: Done.

File: AI-and-Analytics/Features-and-Functionality/IntelTensorFlow_TextGeneration_with_LSTM/README.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+- **Additional Packages**
+
+  You will need to install the additional packages in requirements.txt.
+
+  ```
+  pip install -r requirements.txt
+  ```
+
+### For Intel® DevCloud
```
[the remark this answers is not in the document]
Comment (nedo99) [reply AI-and-Analytics/Features-and-Functionality/IntelTensorFlow_TextGeneration_with_LSTM/README.md]: Done.

**Review by louie-tsai — APPROVED** [review]
looks good. thanks for all the effort.

**Review by jimmytwei — APPROVED** [review]
LGTM. Working on IDC

**Review by jimmytwei — APPROVED** [review]

**Pull request merged**